compiler/qsc/src/interpret.rs

```diff
@@ -292,6 +320,74 @@ impl Interpreter {
         Ok(sim.finish(&val))
     }
 
+    /// Generates a circuit representation for the program.
+    ///
+    /// `entry` can be the current entrypoint, an entry expression, or any operation
+    /// that takes qubits.
+    ///
+    /// An operation can be specified by its name or a lambda expression that only takes qubits.
+    /// e.g. `Sample.Main` , `qs => H(qs[0])`
+    pub fn circuit(
+        &mut self,
+        entry: CircuitEntryPoint,
+    ) -> std::result::Result<Circuit, Vec<Error>> {
+        let mut sink = std::io::sink();
+        let mut out = GenericReceiver::new(&mut sink);
+        let mut sim = CircuitBuilder::new(CircuitConfig {
+            base_profile: self.capabilities.is_empty(),
+        });
+
+        let entry_expr = match entry {
+            CircuitEntryPoint::Operation(operation_expr) => {
+                // To determine whether the passed in expression is a valid callable name
+                // or lambda, we evaluate it and inspect the runtime value.
+                let maybe_operation = self.eval_fragments(&mut out, &operation_expr)?;
+
+                let maybe_invoke_expr = match maybe_operation {
+                    Value::Closure(_, item_id, functor_app)
+                    | Value::Global(item_id, functor_app) => {
+                        // Controlled operations are not supported at the moment.
+                        if functor_app.controlled > 0 {
+                            return Err(vec![Error::NoCircuitForOperation]);
+                        }
+
+                        // Find the item in the HIR
+                        let package = map_fir_package_to_hir(item_id.package);
+                        let local_item_id =
+                            crate::hir::LocalItemId::from(usize::from(item_id.item));
+                        let package_store = self.compiler.package_store();
+
+                        let item = package_store
+                            .get(package)
+                            .and_then(|unit| unit.package.items.get(local_item_id));
+
+                        // Generate the entry expression to invoke the operation.
+                        // Will return `None` if item is not a valid callable that takes qubits.
+                        item.and_then(|item| entry_expr_for_qubit_operation(item, &operation_expr))
+                    }
+                    _ => {
+                        return Err(vec![Error::NoCircuitForOperation]);
+                    }
+                };
+
+                if maybe_invoke_expr.is_none() {
+                    return Err(vec![Error::NoCircuitForOperation]);
+                }
+                maybe_invoke_expr
```

Comment:
`Option<T>` has a method called `ok_or(...)` for this kind of situation. This becomes:

```suggestion
                maybe_invoke_expr.ok_or(vec![Error::NoCircuitForOperation])
```